### ISSUE TITLE
Invalid ID Warning in stress tests

### ISSUE THREAD

**Issue by pro-akim** [wazuh/wazuh#18997]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| from 4.4.4 until 4.6.0 | Wazuh manager - Agent| Manager/Agent | Packages | CentOS |

## Description

It has been detected an unexpected warning in the footprint release testing https://github.com/wazuh/wazuh/issues/17560:

```
Invalid ID 692 for the source ip: '172.31.11.128' (name 'unknown').
```

Research was conducted by QA team in https://github.com/wazuh/wazuh-jenkins/issues/5285

Since `4.4.4-rc2` in footprint VULNERABILITY-DETECTOR-REGISTER test the warning were continuously detected.

While reviewing logs between version 4.4.4 and earlier versions, the following observations were made:

- In earlier versions, there was no noticeable increase in the agent ID throughout the test progression. Specifically, in https://github.com/wazuh/wazuh/issues/17560, a progressive increment of the ID from 004 to 692 was observed in the ossec.log of the manager.

- In version 4.4.2 the log message was not present https://github.com/wazuh/wazuh/issues/16942

- In version 4.4.3, no footprint test was conducted (an issue was opened but closed as "not planned") as indicated in https://github.com/wazuh/wazuh/issues/17252.

- In version 4.3.5, as seen in https://github.com/wazuh/wazuh/issues/14014, log messages from agents 004, 005, and 006 were observed in the ossec.log of the manager.

### Conclusion
It is possible that the observed behavior in version 4.4.4 is related to a modification concerning Windows certificates https://github.com/wazuh/wazuh/blob/v4.4.4/CHANGELOG.md.

It has been considered that it might be related to:

- https://github.com/wazuh/wazuh-jenkins/issues/4867
- https://github.com/wazuh/wazuh/issues/17889
   
However, the reported logs in both cases did not appear during the test, so they have been ruled out. 

Further research is required

**Comment by vikman90** [comment]
Let's check if the agent could connect afterward. If it could, this may be due to a disk load issue but the impact is lower than if the agent could not connect at all.

**Comment by LucioDonda** [comment]
The basic idea of this footprint test is resumed in this python [script](https://github.com/wazuh/wazuh-jenkins/blob/868ebda2cc0a2595a5397ac1d7a837726e44c234/quality/tests/stress/scripts/vuln.py#L22-L23):
* The test consists of a while loop through the whole full load stage.
* Here, as the comment says, each agent registers with the manager, waits a minute and then it's restarted.
![image](https://github.com/wazuh/wazuh/assets/95248059/d02a77de-da26-4391-a7a4-6f57cfc142ae)
* This process takes an average of a minute and 10 seconds on each agent (3 in total).
* Besides that, the registration is done with the **agent-auth** binary, using the manager's IP address, port and a variable hostname on each call:
`popen = subprocess.Popen(authd_command + (manager_ip, '-p', manager_port, '-A', hostname + str(int(total_duration - test_time))), stdout=subprocess.PIPE, stderr=subprocess.PIPE)`
  * The hostname changes with each iteration because it uses the milliseconds of the test_time:
```
  hostname = "{0}-{1}-".format(wh.get_node_name(), random.randint(1,100000))
   hostname + str(int(total_duration - test_time)))
```
                                                            NodeName.......... - RandInt - PendingTestTime
An example of a resultant hostname is: `_C2AMAZ-L273KKC-82563-1461_`

Having said that we can conclude that:
* Using that type of registration method (and changing the hostname intentionally on each one) the manager will respond with a different ID each time. 
* With an approximate calculation of the full load in 42 hours and having 3 agents repeating this process each time, ending the test with agents of IDs near 6000 makes a lot of sense. 
* Besides that and as a double check, the last number of each agent getting registered is the pending time to finish the test and that also makes sense with the above analysis.:
![image](https://github.com/wazuh/wazuh/assets/95248059/7ffb6771-a580-4b7c-bb8c-cc0072722e98)
* As Vikman says in all cases the agent gets successfully registered and the message is printed.
`2023/09/09 12:20:36 agent-auth: INFO: Valid key received`

**Comment by vikman90** [comment]
@LucioDonda That's correct: the agent makes a 20-second delay between the enrollment and the first connection attempt. It does this to let the cluster spread and reload the keys.

Please open an issue to introduce a 10-second delay (at least) in the test (when possible) and close this issue.

**Comment by LucioDonda** [comment]
As @vikman90 said, we have opened an [issue ](https://github.com/wazuh/wazuh-qa/issues/4528).
c.c. @cborla

**Comment by pro-akim** [comment]
## Update
Comment added in:

- https://github.com/wazuh/wazuh-qa/issues/4528

The issue is reopened for further research.
